packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1375,3 +1378,140 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
   .setDescription('Lists all projects that are owned by the caller.')
   .setAction(listProjectsTask)
+
+export const listProposersTask = async (
+  args: { configPath: string },
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { configPath } = args
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  const ChugSplashManager = getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  const proposers = []
+
+  // Fetch current owner
+  const owner = await getProjectOwnerAddress(
+    provider,
+    parsedConfig.options.projectName
+  )
+  proposers.push(owner)
+
+  // Fetch all previous proposers
+  const addProposerEvents = await ChugSplashManager.queryFilter(
+    ChugSplashManager.filters.ProposerAdded()
+  )
+
+  // Verify if each previous proposer is still a proposer before adding it to the list
+  for (const proposerEvent of addProposerEvents) {
+    const address = proposerEvent.args.proposer
+    const isStillProposer = await ChugSplashManager.proposers(address)
+    if (isStillProposer && !proposers.includes(address)) {
+      proposers.push(address)
+    }
+  }
+
+  // Display the list of proposers
+  displayProposerTable(proposers)
+}
+
+task(TASK_CHUGSPLASH_LIST_PROPOSERS)
+  .setDescription('Lists all of the approved proposers for this project')
+  .addPositionalParam(
+    'configPath',
+    'Path to the ChugSplash config file to propose'
+  )
+  .setAction(listProposersTask)
+
+export const addProposerTask = async (
+  args: {
+    configPath: string
+    newProposers: string[]
+  },
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { configPath, newProposers } = args
+
+  if (newProposers.length === 0) {
+    throw new Error('You must specify at least one proposer to add.')
+  }
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+  const provider = hre.ethers.provider
+  const signer = provider.getSigner()
+
+  const spinner = ora()
+
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  const ChugSplashManager = getChugSplashManager(
+    signer,
+    parsedConfig.options.projectName
+  )
+
+  // Fetch current owner
+  const projectOwnerAddress = await getProjectOwnerAddress(
+    provider,
+    parsedConfig.options.projectName
+  )
+  if (projectOwnerAddress !== (await signer.getAddress())) {
+    throw new Error(`Project is owned by: ${projectOwnerAddress}.
+You attempted to add a proposer using address: ${await signer.getAddress()}`)
+  }
+
```

Comment:
If you add a `spinner.start` above, this would be a good place to have a `spinner.succeed`